perf/graaflib/bron_kerbosch_benchmark.cpp

```diff
@@ -0,0 +1,245 @@
+                                                                            30);
+
+template <typename EDGE_T>
+[[nodiscard]] std::vector<graaf::vertex_id_t> create_vertices(
+    graaf::undirected_graph<int, EDGE_T>& graph, size_t n) {
+  std::vector<graaf::vertex_id_t> vertices{};
+  vertices.reserve(n);
+
+  for (size_t i{0}; i < n; ++i) {
+    vertices.push_back(graph.add_vertex(i));
+  }
+
+  return vertices;
+}
+
+template <typename EDGE_T>
+void add_clique(graaf::undirected_graph<int, EDGE_T>& graph,
+                const std::vector<graaf::vertex_id_t>& vertices,
+                size_t start_vertex, size_t clique_size) {
+  // We are in range of vector of vertices
+  size_t end_vertex = std::min(start_vertex + clique_size, vertices.size());
+  std::vector<graaf::edge_id_t> clique{};
+  clique.reserve((clique_size * (clique_size - 1)) / 2);
+
+  // Constructing a clique
+  for (size_t i{start_vertex}; i < end_vertex; ++i) {
+    for (size_t j{i + 1}; j < end_vertex; ++j) {
+      clique.emplace_back(vertices[i], vertices[j]);
+    }
+  }
+
+  for (auto& edge_t : clique) {
+    graph.add_edge(edge_t.first, edge_t.second, 1);
+  }
+}
+}  // namespace
+
+static void bron_kerbosh_two_vertices_cliques(benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  for (auto _ : state) {
+    for (size_t i{0}; i < number_of_edges; ++i) {
+      graph.add_edge(vertices[i], vertices[i + 1], 1);
+    }
+  }
+
+  graaf::algorithm::bron_kerbosch(graph);
+  state.SetComplexityN(state.range(0));
+}
+
+static void bron_kerbosh_dense_graph(benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  for (auto _ : state) {
+    for (size_t i{0}; i < number_of_edges; ++i) {
+      for (size_t j{i + 1}; j < number_of_edges; ++j) {
+        if (i != j && !graph.has_edge(vertices[i], vertices[j]))
+          graph.add_edge(vertices[i], vertices[j], 1);
+      }
+    }
+  }
+
+  graaf::algorithm::bron_kerbosch(graph);
+  state.SetComplexityN(state.range(0));
+}
+
+static void bron_kerbosh_graph_triangle_cliques(benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  size_t clique_size = 4;
```

Comment:
Should the `clique_size` not be equal to 3 if we want to make triangle cliques?

Reply:
My bad with naming, basically,  it's a triangle with a dot inside (4 vertex clique).